File: problems/33/solution.java
// TreeNode type
// class TreeNode {
//     int val;
//     TreeNode left;
//     TreeNode right;
//     TreeNode(int x) { val = x; left = null; right = null; }
// }
class Solution {
    ArrayList<Integer> result = new ArrayList<Integer>();

    public ArrayList<Integer> preorderTraversal(TreeNode root) {
        // Implement the solution
        _inorder(root);
        return result;
    }


    public void _inorder(TreeNode root) {
        if (root == null)
            return;
        result.add(root.val);
        _inorder(root.left);
        _inorder(root.right);
    }

}

File: problems/33/solution.py
class Solution:
    result = []

    def preorder(self, root):
        if not root:
            return
        self.result.append(root.val)
        self.preorder(root.left)
        self.preorder(root.right)


    def preorderTraversal(self, root):
        # Implement the solution here
        self.result = []
        self.preorder(root)
        return self.result
